checkov/kubernetes/graph_builder/graph_components/K8SEdgeBuilder.py

```diff
@@ -0,0 +1,24 @@
+from __future__ import annotations
+from abc import abstractmethod
+
+from checkov.kubernetes.graph_builder.graph_components.blocks import KubernetesBlock
+
+class K8SEdgeBuilder:
+
+    @abstractmethod
+    def should_search_for_edges(self, vertex: KubernetesBlock) -> bool:
+        """
+        implementation should examine vertex's attributes and indicate if it's potentially
+        suitable for the concrete class's edge type.
+        e.g: search for a label attribute in LabelSelectorEdgeBuilder's implementation
+        """
+        raise NotImplementedError
+
+    @abstractmethod
+    def find_connection(self, vertex: KubernetesBlock, vertices: list[KubernetesBlock]) -> list[KubernetesBlock] | None:
```

Comment:
```suggestion
    def find_connection(self, vertex: KubernetesBlock, vertices: list[KubernetesBlock]) -> list[KubernetesBlock]:
```
if we return a list we can return an empty one in case of no connections instead of None